### PR TITLE
Lecture 19 commit homework. adding data class, and fixes

### PR DESCRIPTION
**Pull request by H3lleve**

### CODE REVIEW

**Comment by H3lleve** [comment]
image

File: conftest.py [hunk 1]
```diff
@@ -4,11 +4,38 @@
...
+class SensorInfo:
+    name: str
+    hid: str
+    model: str
+    firmware_version: int
+    reading_interval: int  # RuntimeError: Sensor didn't reset to factory property if float
+
+
+    def __post_init__(self):
+        if not isinstance(self.name, str) or self.name == "":
```
Comment (ddoroshevskyi) [inline conftest.py]: It's good that you're looking for ways to shorten the code. In this case, however, by doing an `or` you removed the distinction between a `TypeError` and a `ValueError`. Generally this is not something we want to do. Same goes for other validations.

File: conftest.py [hunk 2]
```diff
@@ -4,11 +4,38 @@
...
+            raise TypeError("'name' is not a string or empty")
+
+        if not isinstance(self.hid, str) or self.hid == "":
+            raise TypeError("'hid' is not a string or empty")
+
+        if not isinstance(self.model, str) or self.model == "":
+            raise TypeError("'model' is not a string or empty")
+
+        if not isinstance(self.firmware_version, int) or self.firmware_version == "":
```
Comment (ddoroshevskyi) [inline conftest.py]: Testing for a non-empty string is suitable for string properties but here we've got an integer. We'd like to test this for validity from sensor's standpoint. That would be checking that the firmware version is between 10 and 15, both ends included. You could test this with the following code:
```
10 <= self.firmware_version <= 15
```
Of course, you'll need to update the error message in the `ValueError` to reflect this new check.

File: conftest.py [hunk 3]
```diff
@@ -4,11 +4,38 @@
...
+            raise TypeError("'firmware_version' is not an int or empty")
+
+        if not isinstance(self.reading_interval, int) or self.reading_interval == "":
```
Comment (ddoroshevskyi) [inline conftest.py]: Same thing as in my previous comment applies here. We're interested in checking that this is a valid value from sensor's standpoint. That would be seeing that it's equal or more than 1.

File: conftest.py [hunk 4]
```diff
@@ -4,11 +4,38 @@
...
+            raise TypeError("'reading_interval' is not a float or empty")
```
Comment (ddoroshevskyi) [inline conftest.py]: This error message should say `int` instead of `float`.

File: conftest.py [hunk 5]
```diff
@@ -138,7 +165,9 @@ def _make_valid_request(
 def get_sensor_info(make_valid_request):
     def _get_sensor_info():
         log.info("Get sensor info")
-        return make_valid_request(SensorMethod.GET_INFO)
+        sensor_response = make_valid_request(SensorMethod.GET_INFO)
+        return SensorInfo(**sensor_response)
+        # return make_valid_request(SensorMethod.GET_INFO)
```
Comment (ddoroshevskyi) [inline conftest.py]: Better to not have commented code if it's not for explanatory purposes.

**Review by ddoroshevskyi — CHANGES_REQUESTED** [review]

File: conftest.py (hunk 1 above)
Comment (H3lleve) [reply conftest.py]: checks separeted

File: conftest.py (hunk 2 above)
Comment (H3lleve) [reply conftest.py]: fixed

File: conftest.py (hunk 3 above)
Comment (H3lleve) [reply conftest.py]: fixed to         ` if self.reading_interval < 1:`

File: conftest.py (hunk 4 above)
Comment (H3lleve) [reply conftest.py]: fixed

File: conftest.py (hunk 5 above)
Comment (H3lleve) [reply conftest.py]: was left by mistake. fixed

**Pull request merged**